pkg/commands/workload.go

```diff
@@ -324,8 +324,23 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
 		return okToPush, nil
 	}
 
+	var contentDir string
+	if source.IsDir(opts.LocalPath) {
+		contentDir = opts.LocalPath
+	} else if source.IsZip(opts.LocalPath) {
+		zipContentsDir, err := source.HandleZip(opts.LocalPath)
+		if err != nil {
+			c.Errorf("Failed to extract file contents from %q. \n", opts.LocalPath)
+			return false, err
+		}
+		defer os.RemoveAll(zipContentsDir)
+		contentDir = zipContentsDir
+	} else {
+		return false, fmt.Errorf("unsupported file format %q", opts.LocalPath)
```

Comment:
cc @danfein : this is the error if user provides a file which is not a dir and not a valid zip file. If you want something else, please add the msg here